java-checks/src/test/files/checks/ReuseRandomCheck.java

```diff
@@ -0,0 +1,39 @@
+import java.util.Random;
+
+public class A {
+
+  static Random s = new Random(); // Compliant for static field
+  Random f = new Random(); // Compliant for field
+
+  A() {
+    Random v = new Random(); // Compliant in constructor
+  }
+
+  void func(long seed, Random p) {
+    Random a = new Random(); // Noncompliant [[sc=20;ec=26]] {{Save and re-use this "Random".}}
+    Random b = new Random(seed); // Compliant for Random(long seed)
+    Object c = new Object();
+
+    s = new Random();
+    f = new Random();
+    thid.f = new Random();
+
+    f = a = new Random();
+    a = b = new Random(); // false-negative, a and b are local variables, corner case limitation
```

Comment:
I dont' see why we should accept that false negative. Please modify your code so we can handle it.